src/main/java/edu/gwu/csci6212/WildCardChar.java

```diff
@@ -0,0 +1,97 @@
+package edu.gwu.csci6212;
+
+import java.util.*;
+import java.io.*;
+
+public class WildCardChar
+{
+    public static boolean isMatching(char[] c_str, char[] c_pattern, int p, int q)
```

Comment:
It's kind of a taste thing, but I'm not really a bit fan of encodings like "c_", mostly because modern IDEs already can give you all that info just by hovering over the variable name. I'd probably name these parameters something like: sourceString, searchPattern, sourceStringIndex, SearchPatternIndex. I often get tripped up with one letter variables.